src/libs/actions/IOU.ts

```diff
@@ -3157,7 +3160,27 @@ function getUpdateMoneyRequestParams(
         });
     }
 
-    if (policy && PolicyUtils.isPaidGroupPolicy(policy) && updatedTransaction && (hasModifiedTag || hasModifiedCategory || hasModifiedDistanceRate)) {
+    const parentReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${transaction?.reportID}`];
+    if (policy && PolicyUtils.isControlPolicy(policy) && updatedTransaction && hasModifiedCreated && !ReportUtils.isInvoiceReport(parentReport)) {
+        const currentTransactionViolations = allTransactionViolations[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`] ?? [];
+        optimisticData.push(
+            ViolationsUtils.getViolationsOnyxData(
+                updatedTransaction,
+                currentTransactionViolations,
+                policy,
+                policyTagList ?? {},
+                policyCategories ?? {},
+                PolicyUtils.hasDependentTags(policy, policyTagList ?? {}),
+            ),
+        );
+        failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`,
+            value: currentTransactionViolations,
+        });
+    }
```

Comment:
Why is this needed? We don't need to call `ViolationsUtils.getViolationsOnyxData` twice